File: src/2559. Count Vowel Strings in Ranges.js
/**
 * @param {string[]} words
 * @param {number[][]} queries
 * @return {number[]}
 */
function vowelStrings(words, queries) {
  const vowels = ['a', 'e', 'i', 'o', 'u']

  let sum = 0
  const prefixSum = []
  for (let word of words) {
    sum += !!(
      vowels.includes(word[0]) && vowels.includes(word[word.length - 1])
    )
    prefixSum.push(sum)
  }

  const ans = []
  for (let [start, end] of queries) {
    ans.push(prefixSum[end] - (start === 0 ? 0 : prefixSum[start - 1]))
  }
  return ans
}

File: src/2559. Count Vowel Strings in Ranges.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        vowels = {'a', 'e', 'i', 'o', 'u'}

        sum = 0
        prefix_sum = []
        for word in words:
            sum += int(word[0] in vowels and word[-1] in vowels)
            prefix_sum.append(sum)

        ans = []
        for [start, end] in queries:
            ans.append(prefix_sum[end] - (0 if start == 0 else prefix_sum[start - 1]))
        return ans
        